### PR TITLE
Fix docstring for shape of `target` for MultiLabelSoftMarginLoss

### LINKED ISSUE #92000: Fix doc of `torch.nn.MultiLabelSoftMarginLoss`
### 📚 The doc issue

The documentation states that $y[i] \in$ {0, 1} and later on, it also say "label targets padded by -1 ensuring same shape as the input". This is contradictory. Looking at the source code, I believe the bit about padding is incorrect and -1 values will turn the loss the wrong way.

### Suggest a potential alternative/fix

remove "padded by -1"

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### PR DESCRIPTION
**Pull request by dvorst**
Fixes #92000

The documentation at https://pytorch.org/docs/stable/generated/torch.nn.MultiLabelSoftMarginLoss.html#multilabelsoftmarginloss states:
> label targets padded by -1 ensuring same shape as the input.

However, the shape of input and target tensor are compared, and an exception is raised if they differ in either dimension 0 or 1. Meaning the label targets are never padded. See the code snippet below and the resulting output. The documentation is therefore adjusted to:
> label targets must have the same shape as the input.

```
import torch
import torch.nn as nn

# Create some example data
input = torch.tensor(
    [
        [0.8, 0.2, -0.5],
        [0.1, 0.9, 0.3],
    ]
)
target1 = torch.tensor(
    [
        [1, 0, 1],
        [0, 1, 1],
        [0, 1, 1],
    ]
)
target2 = torch.tensor(
    [
        [1, 0],
        [0, 1],
    ]
)
target3 = torch.tensor(
    [
        [1, 0, 1],
        [0, 1, 1],
    ]
)
loss_func = nn.MultiLabelSoftMarginLoss()
try:
    loss = loss_func(input, target1).item()
except RuntimeError as e:
    print('target1 ', e)
try:
    loss = loss_func(input, target2).item()
except RuntimeError as e:
    print('target2 ', e)
loss = loss_func(input, target3).item()
print('target3 ', loss)
```

output:
```
target1  The size of tensor a (3) must match the size of tensor b (2) at non-singleton dimension 0
target2  The size of tensor a (2) must match the size of tensor b (3) at non-singleton dimension 1
target3  0.6305370926856995
```

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -1216,7 +1216,7 @@ class MultiLabelSoftMarginLoss(_WeightedLoss):
 
     Shape:
         - Input: :math:`(N, C)` where `N` is the batch size and `C` is the number of classes.
-        - Target: :math:`(N, C)`, label targets padded by -1 ensuring same shape as the input.
+        - Target: :math:`(N, C)`, label targets must have the same shape as the input.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/loss.py:1219]: Commenting here for posterity: 

The `padded by -1` here seems to have been erroneously copied over from the docs for `MultiLabelMarginLoss` where each row of target is implicitly a list of class indices for the corresponding sample, padded by `-1` if the `num_labels_corresponding_to_sample < C`

https://github.com/pytorch/pytorch/blob/ee4b99cc3a24d2b6ef23bfeb9d4b4afc52e60354/torch/nn/modules/loss.py#L832

This fix is correct as for MultiLabelSoftMarginLoss, every class should be assigned a probability in the target.

**Review by mikaylagawarecki — APPROVED** [review]

**Comment by drisspg** [comment]
After re-reading the docs I think they are actually correct. The `(N,C)` specifies that the label and targets need to be the same shape and the -1 padding refers to the fact that if they are not the same shape you can pad them with -1 in order to get them to be the same shape. Maybe this aspect should be clarified instead?

**Comment by mikaylagawarecki** [comment]
@drisspg Looking at [the implementation](https://github.com/pytorch/pytorch/blob/main/torch/nn/functional.py#L3449) though I don't think this is the case, though it is for `MultiLabelMarginLoss`, which seems to explicitly account for the `-1`s in the kernel?

**Comment by drisspg** [comment]
ohh whoops you are right, I read the wrong one, carry on!

**Comment by dvorst** [comment]
Im a bit confused (I'm kinda new to this) , the label merged is added but the the PR shows up as "closed with un merged commits", will it still be merged later on?

**Comment by mikaylagawarecki** [comment]
@dvorst It has been merged, see [here](https://github.com/pytorch/pytorch/commits/main)

**Comment by dvorst** [comment]
Oh okay, thanks for the quick response! 😁
